music_assistant/server/providers/tidal/__init__.py

```diff
@@ -274,26 +274,29 @@ async def get_library_playlists(self) -> AsyncGenerator[Playlist, None]:
     async def get_album_tracks(self, prov_album_id: str) -> list[Track]:
         """Get album tracks for given album id."""
         tidal_session = await self._get_tidal_session()
-        return [
-            await self._parse_track(track_obj=track)
-            for track in await get_album_tracks(tidal_session, prov_album_id)
-        ]
+        try:
+            album_tracks_obj = await get_album_tracks(tidal_session, prov_album_id)
+        except MediaNotFoundError as err:
+            raise err
```

Comment:
What is the point here as this just raises the same exception without any side effect ?

Reply:
Should I just remove the "as err:" and simply raise?

So:

` try:
            album_tracks_obj = await get_album_tracks(tidal_session, prov_album_id)
        except MediaNotFoundError:
            raise`
Still trying to fully get to grips with python exception handling.

Reply:
Well, you are catching MediaNotFoundError and raise it without doing anything else, there is no need to catch it at all, just let it bubble up so remove the entire try..except clause here.

catch and reraise only if you want to intervene, like doing something with the data or raise another exception etc.

Reply:
I could have sworn that during my testing, if I didn't catch and re-raise it would cause the sync to fail, but I just removed all the extraneous catches and everything is working as expected.